### PR TITLE
Add native implementation of inv (Gauss-Jordan Elimination)

### PR DESCRIPTION
**Pull request by rahulghangas**
- [ ] Add inplace implementation
- [x] Testing

### CODE REVIEW

**Comment by rahulghangas** [comment]
@ben-albrecht needs label gsoc:LinearAlgebra

File: modules/packages/LinearAlgebra.chpl [hunk 1]
```diff
@@ -806,6 +806,70 @@ proc inv (ref A: [?Adom] ?eltType, overwrite=false) where usingLAPACK {
   return A;
 }
 
+pragma "no doc"
+proc inv(ref A: [?Adom] ?eltType) throws
+where !usingLAPACK
+{
+  if Adom.rank != 2 then
+    halt("Wrong rank for matrix inverse");
```
Comment (ben-albrecht) [inline modules/packages/LinearAlgebra.chpl:814]: This can be a `compilerError`

File: modules/packages/LinearAlgebra.chpl [hunk 2]
```diff
@@ -806,6 +806,70 @@ proc inv (ref A: [?Adom] ?eltType, overwrite=false) where usingLAPACK {
...
+
+  if !isSquare(A) then
+    halt("Matrix inverse only supports square matrices");
```
Comment (ben-albrecht) [inline modules/packages/LinearAlgebra.chpl:817]: This can be a compilerError

File: modules/packages/LinearAlgebra.chpl (hunk 1 above)
Comment (ben-albrecht) [reply modules/packages/LinearAlgebra.chpl:814]: I believe `isSquare` covers this requirement as well, so we could drop this check.

File: modules/packages/LinearAlgebra.chpl [hunk 3]
```diff
@@ -806,6 +806,70 @@ proc inv (ref A: [?Adom] ?eltType, overwrite=false) where usingLAPACK {
...
+
+  const n = A.shape(0);
+  var ACopy : [1..n, 1..n] real = [x in A] x;
```
Comment (ben-albrecht) [inline modules/packages/LinearAlgebra.chpl:820]: Does this not work?

```suggestion
  var ACopy = A;
```

File: modules/packages/LinearAlgebra.chpl [hunk 4]
```diff
@@ -806,6 +806,70 @@ proc inv (ref A: [?Adom] ?eltType, overwrite=false) where usingLAPACK {
...
+
+  private use Math;
```
Comment (ben-albrecht) [inline modules/packages/LinearAlgebra.chpl:822]: `Math` is always used by default.

File: modules/packages/LinearAlgebra.chpl (hunk 2 above)
Comment (rahulghangas) [reply modules/packages/LinearAlgebra.chpl:917]: compilerError doesn't work if the argument to inv is a call to another function that returns a 2-D matrix. halt does

File: modules/packages/LinearAlgebra.chpl [hunk 5]
```diff
@@ -901,6 +901,73 @@ proc inv (ref A: [?Adom] ?eltType, overwrite=false) where usingLAPACK {
   return A;
 }
 
+/*
+  Returns the inverse of ``A`` square matrix A.
+
+
+    .. note::
+
+      This procedure is used when `lacpakImpl=off` is used and currently
```
Comment (ben-albrecht) [inline modules/packages/LinearAlgebra.chpl:910]:
```suggestion
      This procedure is used when `lapackImpl=off` is used and currently
```

File: modules/packages/LinearAlgebra.chpl (hunk 2 above)
Comment (ben-albrecht) [reply modules/packages/LinearAlgebra.chpl:917]: That seems like a bug in the compiler to me... Do you have a short example demonstrating this?

**Comment by e-kayrakli** [comment]
@rahulghangas 

> Add inplace implementation

Is this still a todo? If not, and this is ready for a review, could you update the PR message and let us know, so we move forward with this.

If it is still a todo let us know if you can/want to work on it soon.

**Comment by e-kayrakli** [comment]
Based on silence, I am closing this PR.

@rahulghangas -- Please feel free to reopen when you have the time to work on this.

**Pull request closed without merging**